File: Frontend/src/app/features/events/model/registration-response.ts
import { FoodPreference } from '@features/events/model/food-preference';
import { RegistrationStatus } from '@features/events/model/registration-status';

export type RegistrationResponse = {
  id: number;
  eventId: number;
  userId: number;
  registrationDate: string;
  gdprConsent: boolean;
  photoConsent: boolean;
  foodPreference: FoodPreference | null;
  transportNeeded: boolean | null;
  driverName: string | null;
  driverPhoneNumber: string | null;
  accommodationNeeded: boolean | null;
  accommodationDays: number | null;
  status: RegistrationStatus;
};

File: Backend/src/main/java/com/example/CheckInApp/dto/response/RegistrationResponse.java
package com.example.CheckInApp.dto.response;

import com.example.CheckInApp.model.FoodPreference;
import com.example.CheckInApp.model.RegistrationStatus;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.Instant;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class RegistrationResponse {

    private Long id;
    private Long eventId;
    private Long userId;
    private Instant registrationDate;
    private Boolean gdprConsent;
    private Boolean photoConsent;
    private FoodPreference foodPreference;
    private Boolean transportNeeded;
    private String driverName;
    private String driverPhoneNumber;
    private Boolean accommodationNeeded;
    private Integer accommodationDays;
    private RegistrationStatus status;

}
